brightStars_DM_sprint_Feb2020/BeyondRadius.ipynb

```diff
@@ -0,0 +1,422 @@
+    "src_loaded = False "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Initialize Butler repo:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "butler = Butler(butlerPath)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Path to masks. Note there's likely some that are repeated when using Arcturus near the edges of patches."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "r\n"
+     ]
+    }
+   ],
+   "source": [
+    "if gouldingMask:\n",
+    "    # filenames have lower case filter in them:\n",
+    "    if chosenFilter=='HSC-G':\n",
+    "        filter_lc = 'g'\n",
+    "    elif chosenFilter=='HSC-R':\n",
+    "        filter_lc = 'r'\n",
+    "    elif chosenFilter=='HSC-I':\n",
+    "        filter_lc = 'i'\n",
+    "    elif chosenFilter=='HSC-z':\n",
+    "        filter_lc = 'z'\n",
+    "    elif chosenFilter=='HSC-y':\n",
+    "        filter_lc = 'y'\n",
+    "    print(filter_lc)\n",
+    "    mask_path = \"/datasets/hsc/BrightObjectMasks/GouldingMasksS18A/new_S18Amask_{}.reg\".format(\n",
+    "                                            filter_lc)\n",
+    "else:\n",
+    "    mask_path = \"/datasets/hsc/BrightObjectMasks/ArcturusMasks/\" "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If plots are to be saved, make sure the folder exists:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if savePlots:\n",
+    "    if not os.path.isdir('plots/'):\n",
+    "        os.makedirs('plots/')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run\n",
+    "Warning: running cell below can take a while."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "densities, brightMags, brightRads, brightCenters = [], [], [], []\n",
+    "for tract,patches in zip(tractList,patchList):\n",
+    "    # get sky map and WCS for this tract\n",
+    "    skymap = butler.get(\"deepCoadd_skyMap\", tract=tract, filter=chosenFilter)\n",
+    "    tractInfo = skymap[tract]\n",
+    "    wcs = tractInfo.getWcs()\n",
+    "    \n",
+    "    # convert annuli parameters to pixel\n",
+    "    pixScale = geom.radToArcsec(wcs.getPixelScale())\n",
+    "    annSizePix = annSize / pixScale\n",
+    "    \n",
+    "    # load detection catalog for each patch\n",
+    "    if verbose:\n",
+    "        print(\"Loading and appending all detections...\")\n",
+    "    if not src_loaded:\n",
+    "        src = butler.get('deepCoadd_meas', tract=tract, patch=patches[0], \n",
+    "                         filter=chosenFilter)\n",
```

Comment:
To save time/memory, you can add a flag to this get if you are not making use of the object footprints:
`flags=afwTable.SOURCE_IO_NO_FOOTPRINTS` (omit them entirely) or `flags=afwTable.SOURCE_IO_NO_HEAVY_FOOTPRINTS` (just omit the "heavy" parts).

Reply:
(Again, thank you _so_ much for this tip!)